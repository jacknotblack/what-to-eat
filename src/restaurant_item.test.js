import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import RestaurantItem from "./restaurant_item";

describe("normal suite", () => {
    let wrapper;
    const props = {
        name: "testName",
        type: "testType"
    };

    beforeEach(() => {
        wrapper = shallow(<RestaurantItem {...props} />);
    });

    describe('rendering', () => {
        it("renders without crashing", () => {
            expect(wrapper.length).toEqual(1);
        });

        it("should match snapshot", () => {
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it("should render name and type", () => {
            expect(wrapper.find(".restaurantItem").text()).toEqual("Type: testType Name: testName");

        });

    })

});
