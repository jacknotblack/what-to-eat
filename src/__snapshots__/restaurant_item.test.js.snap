// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normal suite rendering should match snapshot 1`] = `
<div
  className="restaurantItem"
>
  Type: 
  testType
   
  <br />
  Name: 
  testName
  <br />
  <br />
</div>
`;
